core/test/financial-templates/PricelessPositionManager.js

```diff
@@ -974,4 +982,178 @@ contract("PricelessPositionManager", function(accounts) {
     collateralPaid = (await collateral.balanceOf(other)).sub(initialCollateral);
     assert.equal(collateralPaid, toWei("120"));
   });
+
+  it("Emergency shutdown: lifecycle", async function() {
+    // To mock the emergency shutdown, register a controlled EOA as the `Governor` within the `Finder`.
+    const mockFinancialContractsAdmin = web3.utils.utf8ToHex("FinancialContractsAdmin");
+    await finder.changeImplementationAddress(mockFinancialContractsAdmin, mockGovernor, {
+      from: contractDeployer
+    });
+
+    // Create one position with 100 synthetic tokens to mint with 150 tokens of collateral. For this test say the
+    // collateral is Dai with a value of 1USD and the synthetic is some fictional stock or commodity.
+    await collateral.approve(pricelessPositionManager.address, toWei("100000"), { from: sponsor });
+    const numTokens = toWei("100");
+    const amountCollateral = toWei("150");
+    await pricelessPositionManager.create({ rawValue: amountCollateral }, { rawValue: numTokens }, { from: sponsor });
+
+    // Transfer half the tokens from the sponsor to a tokenHolder. IRL this happens through the sponsor selling tokens.
+    const tokenHolderTokens = toWei("50");
+    await tokenCurrency.transfer(tokenHolder, tokenHolderTokens, {
+      from: sponsor
+    });
+
+    // Some time passes and the UMA token holders decide that Emergency shutdown needs to occur.
+    const shutdownTimestamp = expirationTimestamp - 1000;
+    await pricelessPositionManager.setCurrentTime(shutdownTimestamp);
+
+    // Should revert if emergency shutdown initialized by non-FinancialContractsAdmin (governor).
+    assert(
+      await didContractThrow(
+        pricelessPositionManager.emergencyShutdown({
+          from: other
+        })
+      )
+    );
+
+    // FinancialContractAdmin can initiate emergency shutdown.
+    const emergencyShutdownResult = await pricelessPositionManager.emergencyShutdown({ from: mockGovernor });
+
+    // Check the event returned the correct values.
+    truffleAssert.eventEmitted(emergencyShutdownResult, "EmergencyShutdown", ev => {
+      return (
+        ev.caller == mockGovernor &&
+        ev.originalExpirationTimestamp == expirationTimestamp &&
+        ev.shutdownTimestamp == shutdownTimestamp
+      );
+    });
+
+    // Check contract state change correctly to requested oracle price and the contract expiration has updated.
+    assert.equal(await pricelessPositionManager.contractState(), STATES.EXPIRED_PRICE_REQUESTED);
+    assert.equal((await pricelessPositionManager.expirationTimestamp()).toString(), shutdownTimestamp.toString());
+
+    // Emergency shutdown should not be able to be called a second time.
+    assert(
+      await didContractThrow(
+        pricelessPositionManager.emergencyShutdown({
+          from: mockGovernor
+        })
+      )
+    );
+
+    // Before the DVM has resolved a price withdrawals should be disabled (as with settlement at maturity).
```

Comment:
Perhaps add a test to ensure that `expire()` can't be called either

Reply:
sure. done